pkg/runtime/grpc/grpc-runtime.go

```diff
@@ -54,33 +70,44 @@ const (
 )
 
 type Runtime struct {
-	info          *deployinfo.DeployInfo
-	defaultValues map[string]string
+	info           *deployinfo.DeployInfo
+	defaultValues  map[string]string
+	globalParams   *params.Params
+	connectionMode ConnectionMode
+}
+
+type RunClient interface {
+	Recv() (*api.GadgetEvent, error)
 }
 
 // New instantiates the runtime and loads the locally stored gadget info. If no info is stored locally,
 // it will try to fetch one from one of the gadget nodes and store it locally. It will issue warnings on
 // failures.
-func New(skipInfo bool) *Runtime {
+func New(options ...Option) *Runtime {
 	r := &Runtime{
 		defaultValues: map[string]string{},
 	}
-
-	if skipInfo {
-		return r
+	for _, option := range options {
+		option(r)
 	}
+	return r
+}
 
+// InitInfo loads the locally stored gadget catalog. If no catalog is stored locally,
+// it will try to fetch one from one of the gadget nodes and store it locally. It will issue warnings on
+// failures.
+func (r *Runtime) InitInfo(params *params.Params) error {
```

Comment:
"It will issue warnings on failures": Should the return `error` be removed then?